Automattic/jetpack
# Settings: Improve Instant Search toggle UX

Fixes #19386. Reported originally by @atanas-dev.

#### Changes proposed in this Pull Request:
* Changes the toggle behavior for the Search card in the Performance settings section.
  * Before: Instant Search toggle would be disabled if the search module was disabled.
  * After: Instant Search toggle is always enabled. If toggled on while the search module is disabled, the search module will also be toggled on.

This fixes a bug where the Instant Search toggle would flip on and off when disabling both Search and Instant Search settings via the Search toggle.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change on your site with a Jetpack Search subscription.
* Navigate to `/wp-admin/admin.php?page=jetpack#/performance`.
* Try toggling the Search module toggle and the Instant Search setting toggle. Ensure that they all work as expected. For example:
  a) With both toggles disabled, toggle on the search module.
  b) With both toggles disabled, toggle on the Instant Search setting.
  c) With just the search module enabled, toggle off the search module.
